### PR TITLE
Tune everything + clean up.

### PR DESCRIPTION
**Pull request by amackworth**

### CODE REVIEW

File: QuasarHelios2015/src/org/team1540/quasarhelios/Elevator.java [hunk 1]
```diff
@@ -33,7 +34,7 @@ public static void setup() {
         lowering.toggleWhen(loweringInput);
         lowering.setFalseWhen(raisingInput);
 
-        FloatMixing.pumpWhen(QuasarHelios.globalControl, Mixing.quadSelect(raising, lowering, 0.0f, -1.0f, 1.0f, 0.0f), winch);
+        FloatMixing.pumpWhen(QuasarHelios.globalControl, Mixing.quadSelect(raising, lowering, 0.0f, -winchSpeed.get(), winchSpeed.get(), 0.0f), winch);
```
Comment (celskeggs) [inline QuasarHelios2015/src/org/team1540/quasarhelios/Elevator.java]: Think about what you're doing here...

File: QuasarHelios2015/src/org/team1540/quasarhelios/Rollers.java [hunk 2]
```diff
@@ -22,8 +22,8 @@
 	
 	private static final FloatOutput externalRollers = FloatMixing.combine(armRollers, frontRollers);
 	private static final FloatOutput allRollers = FloatMixing.combine(externalRollers, internalRollers);
-	
-	private static final FloatInput motorSpeed = FloatMixing.createDispatch(Mixing.quadSelect(running, direction, 0.0f, 0.0f, -1.0f, 1.0f), Igneous.globalPeriodic);
+	private static final FloatInputPoll actualSpeed = ControlInterface.mainTuning.getFloat("main-rollers-speed", 1.0f);
+	private static final FloatInput motorSpeed = FloatMixing.createDispatch(Mixing.quadSelect(running, direction, 0.0f, 0.0f, -actualSpeed.get(), actualSpeed.get()), Igneous.globalPeriodic);
```
Comment (celskeggs) [inline QuasarHelios2015/src/org/team1540/quasarhelios/Rollers.java]: Same thing here...

File: QuasarHelios2015/src/org/team1540/quasarhelios/Elevator.java (hunk 1 above)
Comment (celskeggs) [inline QuasarHelios2015/src/org/team1540/quasarhelios/Elevator.java]: Think about program phases.

Comment (celskeggs) [inline QuasarHelios2015/src/org/team1540/quasarhelios/Elevator.java]: Compiling, setup, running.

**Comment by AidanSmith** [comment]
If you are moving all tuning contexts to control interface fields, I added some tuning in DriveCode.java that you could move over to control interface.

File: QuasarHelios2015/src/org/team1540/quasarhelios/Elevator.java [hunk 3]
```diff
@@ -33,7 +34,7 @@ public static void setup() {
         lowering.toggleWhen(loweringInput);
         lowering.setFalseWhen(raisingInput);
 
-        FloatMixing.pumpWhen(QuasarHelios.globalControl, Mixing.quadSelect(raising, lowering, 0.0f, -1.0f, 1.0f, 0.0f), winch);
+        FloatMixing.pumpWhen(QuasarHelios.globalControl, Mixing.select(raising, Mixing.select(lowering, FloatMixing.always(0.0f), FloatMixing.createDispatch(winchSpeed, Igneous.globalPeriodic)), Mixing.select(lowering, FloatMixing.always(0.0f), FloatMixing.negate(FloatMixing.createDispatch(winchSpeed, Igneous.globalPeriodic)))), winch);
```
Comment (celskeggs) [inline QuasarHelios2015/src/org/team1540/quasarhelios/Elevator.java]: You don't need `createDispatch`.

File: QuasarHelios2015/src/org/team1540/quasarhelios/Rollers.java [hunk 4]
```diff
@@ -22,8 +22,8 @@
 	
 	private static final FloatOutput externalRollers = FloatMixing.combine(armRollers, frontRollers);
 	private static final FloatOutput allRollers = FloatMixing.combine(externalRollers, internalRollers);
-	
-	private static final FloatInput motorSpeed = FloatMixing.createDispatch(Mixing.quadSelect(running, direction, 0.0f, 0.0f, -1.0f, 1.0f), Igneous.globalPeriodic);
+	private static final FloatInputPoll actualSpeed = ControlInterface.mainTuning.getFloat("main-rollers-speed", 1.0f);
+	private static final FloatInput motorSpeed = FloatMixing.createDispatch(Mixing.select(running, Mixing.select(direction, FloatMixing.createDispatch(actualSpeed, Igneous.globalPeriodic), FloatMixing.createDispatch(FloatMixing.negate(actualSpeed), Igneous.globalPeriodic)), Mixing.select(direction, FloatMixing.always(0.0f), FloatMixing.always(0.0f))), Igneous.globalPeriodic);
```
Comment (celskeggs) [inline QuasarHelios2015/src/org/team1540/quasarhelios/Rollers.java]: Same as above.

File: QuasarHelios2015/src/org/team1540/quasarhelios/Rollers.java [hunk 5]
```diff
@@ -22,8 +21,8 @@
 	
 	private static final FloatOutput externalRollers = FloatMixing.combine(armRollers, frontRollers);
 	private static final FloatOutput allRollers = FloatMixing.combine(externalRollers, internalRollers);
-	
-	private static final FloatInput motorSpeed = FloatMixing.createDispatch(Mixing.quadSelect(running, direction, 0.0f, 0.0f, -1.0f, 1.0f), Igneous.globalPeriodic);
+	private static final FloatInput actualSpeed = ControlInterface.mainTuning.getFloat("main-rollers-speed", 1.0f);
+	private static final FloatInput motorSpeed = Mixing.select(running, Mixing.select(direction, actualSpeed, FloatMixing.negate(actualSpeed)), Mixing.select(direction, FloatMixing.always(0.0f), FloatMixing.always(0.0f)));
```
Comment (celskeggs) [inline QuasarHelios2015/src/org/team1540/quasarhelios/Rollers.java]: Any particular reason that you're selecting between 0.0 and 0.0?

**Comment by celskeggs** [comment]
Looks good.

**Pull request merged**